modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -579,20 +562,19 @@ FeatureFloodCount::mergeSets(bool use_periodic_boundary_info)
   } // map loop
 
 
-  for (unsigned int map_num = 0; map_num < _maps_size; ++map_num)
+  for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
     _feature_sets[map_num].clear();
 
   // Now consolidate the data structure
   _feature_count = 0;
-  for (processor_id_type rank = 0; rank < n_procs; ++rank)
-    for (unsigned int map_num = 0; map_num < _maps_size; ++map_num)
+  for (auto rank = decltype(_n_procs)(0); rank < _n_procs; ++rank)
+    for (auto map_num = decltype(_maps_size)(0); map_num < _maps_size; ++map_num)
     {
-      for (std::vector<FeatureData>::iterator it = _partial_feature_sets[rank][map_num].begin();
-           it != _partial_feature_sets[rank][map_num].end(); ++it)
+      for (auto & feature : _partial_feature_sets[rank][map_num])
       {
-        if (!it->_merged)
+        if (!feature._merged)
         {
-          _feature_sets[map_num].push_back(MooseSharedPointer<FeatureData>(new FeatureData(*it)));
+          _feature_sets[map_num].emplace_back(std::unique_ptr<FeatureData>(new FeatureData(feature)));
```

Comment:
BTW: Despite what I said below about how to use `emplace_back()` this _is_ the best you can do for putting pointers into a `std::vector` using C++11.  Even in C++14 you still want to wrap the pointer in `make_unique()` like so:

```C++
_feature_sets[map_num].emplace_back(std:make_unique<FeatureData>(feature));
```

Either way you _don't_ want to try passing a raw pointer in to be forwarded like this:

```C++
_feature_sets[map_num].emplace_back(new FeatureData(feature));
```

The reason is that the pointer will be "discarded" if the vector throws an exception... which means it will leak memory.